src/ConnectedMode/Migration/FixedStepsProgressToMigrationProgressConverter.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarLint.VisualStudio.ConnectedMode.Binding;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    internal class FixedStepsProgressToMigrationProgressConverter : IProgress<FixedStepsProgress>
+    {
+        private readonly IProgress<MigrationProgress> progress;
+
+        public FixedStepsProgressToMigrationProgressConverter(IProgress<MigrationProgress> progress)
+        {
+            this.progress = progress;
+        }
+
+        public void Report(FixedStepsProgress value)
+        {
+            progress?.Report(new MigrationProgress(0, 1, value.Message, false));
```

Comment:
Right now we really only use the value.message part which makes this simple to convert.